code/modules/reagents/reagent_containers/bidon.dm

```diff
@@ -0,0 +1,172 @@
+		reagents.add_reagent(starting_reagent, starting_volume)
+
+/obj/structure/reagent_dispensers/bidon/advanced
+	name = "stasis B.I.D.O.N. canister"
+	desc = "An advanced B.I.D.O.N. canister with stasis function."
+	icon_state = "bidon_adv"
+	atom_flags = ATOM_REAGENTS_SKIP_REACTIONS // It's a stasis BIDON, shouldn't allow chems to react inside it.
+	filling_states = list(20,40,60,80,100)
+	volume = 9000
+
+/obj/structure/reagent_dispensers/bidon/trigger
+	name = "trigger-stasis B.I.D.O.N. canister"
+	desc = "An advanced B.I.D.O.N. canister with a stasis function that can be temporarily disabled with a multitool."
+	icon_state = "bidon_adv"
+	atom_flags = ATOM_REAGENTS_SKIP_REACTIONS //Tho its not a subtype its meant to be
+	filling_states = list(20,40,60,80,100)
+	volume = 9000
+	var/timer_till_mixing = 120
+	var/timing = FALSE
+
+/obj/structure/reagent_dispensers/bidon/trigger/examine(mob/user)
+	. = ..()
+	if(timing)
+		to_chat(user, SPAN_DANGER("[timer_till_mixing] seconds until stasis is disabled."))
+	else
+		to_chat(user, SPAN_NOTICE("[src]'s timer isn't activated."))
+
+/obj/structure/reagent_dispensers/bidon/trigger/attackby(obj/item/I, mob/user)
+	if(!timing)
+		if(I.is_multitool())
+			timing=TRUE
+			to_chat(user, SPAN_NOTICE("You start the timer."))
+			spawn(10 * timer_till_mixing)
+			timer_end()
+			timing=FALSE
+			return
+	else
+		. = ..()
+	update_icon()
+
+/obj/structure/reagent_dispensers/bidon/trigger/proc/timer_end()
+	atom_flags &= ~(ATOM_REAGENTS_SKIP_REACTIONS)
+	spawn(10)
+	atom_flags |= ATOM_REAGENTS_SKIP_REACTIONS
+	reagents.handle_reactions()
+
+/obj/structure/reagent_dispensers/bidon/Initialize(mapload, ...)
+	. = ..()
+	update_icon()
+
+/obj/structure/reagent_dispensers/bidon/examine(mob/user)
+	. = ..()
+	if(get_dist(user, src) <= 2)
+		if(lid)
+			to_chat(user, SPAN_NOTICE("It has lid on it."))
+		if(reagents.total_volume)
+			to_chat(user, SPAN_NOTICE("It's filled with [reagents.total_volume]/[volume] units of reagents."))
+
+/obj/structure/reagent_dispensers/bidon/attack_hand(mob/user as mob)
+	//Prevent the bidon from being messed with while it is anchored.
+	if(anchored)
+		to_chat(user, SPAN_NOTICE("You can't remove the lid while the canister is anchored!"))
+		return
+	lid = !lid
+	if(lid)
+		to_chat(user, SPAN_NOTICE("You put the lid on."))
+		atom_flags &= ~(ATOM_REAGENTS_IS_OPEN)
+	else
+		atom_flags |= ATOM_REAGENTS_IS_OPEN
+		to_chat(user, SPAN_NOTICE("You removed the lid."))
+	playsound(src,'sound/items/trayhit2.ogg',50,1)
+	update_icon()
+
+/obj/structure/reagent_dispensers/bidon/attackby(obj/item/I, mob/user)
+
+	//Handle attaching the BIDON to a valid machine, should one exist
+	if(I.is_wrench())
+		if(anchored)
+			anchored = FALSE
+			anchored_machine = null
+			playsound(src, I.usesound, 50, 1)
+		else
+			var/list/directions = list(WEST, NORTH, SOUTH, EAST)
+			for(var/direction_from_obj in directions)
+				for (var/obj/machinery/valid_machine in get_step(get_turf(src), direction_from_obj))
```

Comment:
```suggestion
			for (var/obj/machinery/valid_machine in orange(1))
```
`anchor_direction` is bound to cardinals, and `reverse_direction()` respects diagonals, so the equality will much more concisely exclude diagonal candidates.